esphome/components/uart/__init__.py

```diff
@@ -94,7 +96,29 @@ def _uart_declare_type(value):
     "BOTH": UARTDirection.UART_DIRECTION_BOTH,
 }
 
-AFTER_DEFAULTS = {CONF_BYTES: 256, CONF_TIMEOUT: "100ms"}
+# The reason for having CONF_BYTES at 150 by default:
+#
+# The log message buffer size is 512 bytes by default. About 35 bytes are
+# used for the log prefix. That leaves us with 477 bytes for logging data.
+# The default log output is hex, which uses 3 characters per represented
+# byte (2 hex chars + 1 separator). That means that 477 / 3 = 159 bytes
+# can be represented in a single log line. Using 150, because people love
+# round numbers.
+AFTER_DEFAULTS = {CONF_BYTES: 150, CONF_TIMEOUT: "100ms"}
+
+# By default, log in hex format when no specific sequence is provided.
+DEFAULT_DEBUG_OUTPUT = "UARTDebug::log_hex(direction, bytes, ':');"
+DEFAULT_SEQUENCE = [{CONF_LAMBDA: make_data_base(DEFAULT_DEBUG_OUTPUT)}]
+
+
+def maybe_empty_debug(schema):
+    def validator(value):
+        if value is None:
+            value = {}
+        return schema(value)
+
+    return validator
```

Comment:
Can slim this down a bit. (Along with the next change too)
```suggestion
def maybe_empty_debug(value):
    if value is None:
        value = {}
    return DEBUG_SCHEMA(value)
```

Reply:
Didn't know if the function factory style was the preferred way to go. Now I know 🙂 Thanks.